substrate/client/rpc-spec-v2/src/chain_head/tests.rs

```diff
@@ -3833,3 +3833,222 @@ async fn follow_unique_pruned_blocks() {
 	});
 	assert_eq!(event, expected);
 }
+
+#[tokio::test]
+async fn follow_report_best_block_of_a_known_block() {
```

Comment:
Are we sure this is testing what was changed here? I wanted to understand this better and ran the test on master to see it fail, but it passed there too 😬

Reply:
Yep that's correct. Initially, I wanted to check by this test the fact assumption that we can generate multiple BestBlock events (even for blocks that are descendant of the current best block), similar to:
` Best 4, Best 5, Best 4, Finalized 4`.

It turned out to not be as complicated as I thought initially, and have modified the testing to use the internal client mock of chainHead. The test should now cover the exact race condition of `self.client.info()` and fail on origin/master :D Thanks for pointing this out 🙏